hail/python/hail/expr/aggregators/aggregators.py

```diff
@@ -1508,13 +1508,52 @@ def linreg(y, x, nested_dim=1, weight=None) -> StructExpression:
 
     hl.methods.statgen._warn_if_no_intercept('linreg', x)
 
-    if weight is None:
-        return _linreg(y, x, nested_dim)
-    else:
-        return _linreg(hl.sqrt(weight) * y,
-                       [hl.sqrt(weight) * xi for xi in x],
-                       nested_dim)
+    if weight is not None:
+        y = hl.sqrt(weight) * y
+        x = [hl.sqrt(weight) * xi for xi in x]
+    temp = _linreg(y, x, nested_dim)
 
+    k = len(x)
+    k0 = nested_dim
+    n = hl.agg.count_where(hl.all(lambda x: hl.is_defined(x), x) & hl.is_defined(y))
+    yty = hl.agg.filter(hl.is_defined(x), hl.agg.sum(y*y))
```

Comment:
what is `hl.is_defined(x)`? Is that an `is_defined` of the implicitly created MakeArray of the `x` exprs? 🤔

Reply:
I think `n` and `yty` should both be inside the same `hl.agg.filter` above, whose result is the count and the sum_ysq